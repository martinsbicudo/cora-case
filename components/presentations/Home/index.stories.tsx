import { StoryObj, Meta } from '@storybook/react'

import Home from '.'

export default {
  title: 'Presentations / Home',
  component: Home,
} as Meta

export const Default: StoryObj = {}
