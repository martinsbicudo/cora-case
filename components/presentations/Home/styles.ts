import styled from 'styled-components'

export const Home = styled.div`
  padding-bottom: 40px;
`

export const HomeLoadingBox = styled.div`
  padding: 50px 0;
`
