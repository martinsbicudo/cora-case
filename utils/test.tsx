import type { ReactElement, ReactNode } from 'react'

import { ThemeProvider } from 'styled-components'

import theme from '@/styles/theme'
import { render, RenderOptions } from '@testing-library/react'

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
export { customRender as render }
