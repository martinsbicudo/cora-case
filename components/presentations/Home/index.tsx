import Head from 'next/head'

import { Container } from '@Commons'
import { Header, Filters } from '@Containers'

import * as Styled from './styles'

const Home = () => {
  return (
    <>
      <Head>
        <title>Cora Case</title>
        <meta name="description" content="Cora Case" />
        <link rel="icon" href="favicon.png" />
      </Head>
      <Styled.Home>
        <Header />
        <main>
          <Container>
            <Filters />
          </Container>
        </main>
      </Styled.Home>
    </>
  )
}

export default Home
