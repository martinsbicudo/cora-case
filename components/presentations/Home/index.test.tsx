import { render } from '@/test/helpers'

import Home from '.'

describe('Presentations / Home', () => {
  const renderComponent = () => render(<Home />)

  it('should render component', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
