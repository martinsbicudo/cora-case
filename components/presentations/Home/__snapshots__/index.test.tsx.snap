// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Presentations / Home should render component 1`] = `
<div>
  <div
    class="sc-ipEyDJ"
  >
    <header
      class="sc-jrcTuL AEVqD"
    >
      <div
        class="sc-gswNZR iKhUPL"
      >
        <h1
          class="sc-kDvujY kndZqW"
        >
          Extrato
        </h1>
      </div>
    </header>
    <main>
      <div
        class="sc-gswNZR iKhUPL"
      >
        <section
          class="sc-pyfCe hBEMOj"
        >
          <fieldset
            class="sc-hLBbgP buESsb"
          >
            <div
              class="sc-eDvSVe HnORa"
            >
              <i>
                icon-mock
              </i>
            </div>
            <input
              class="sc-jSUZER gjFiss"
              id="search"
              name="search"
              placeholder="Pesquisar"
              type="text"
            />
          </fieldset>
          <ul
            class="sc-gKPRtg jTTWzr"
          >
            <li
              class="sc-iBYQkv dETHzI"
            >
              <button
                class="sc-ftTHYK jllEQx"
                type="button"
              >
                Tudo
              </button>
            </li>
            <li
              class="sc-iBYQkv dETHzI"
            >
              <button
                class="sc-ftTHYK bVbavl"
                type="button"
              >
                Entrada
              </button>
            </li>
            <li
              class="sc-iBYQkv dETHzI"
            >
              <button
                class="sc-ftTHYK bVbavl"
                type="button"
              >
                Sa√≠da
              </button>
            </li>
            <li
              class="sc-iBYQkv dETHzI"
            >
              <button
                class="sc-ftTHYK bVbavl"
                type="button"
              >
                Futuro
              </button>
            </li>
          </ul>
        </section>
      </div>
    </main>
  </div>
</div>
`;
